src/Orchard.Web/Modules/Orchard.Comments/Drivers/CommentsPartDriver.cs

```diff
@@ -77,6 +77,22 @@ public class CommentsPartDriver : ContentPartDriver<CommentsPart> {
                         return shapeHelper.Parts_Comments_Count(
                             CommentCount: part.CommentsCount);
                     }),
+                ContentShape("Parts_Comments_Count_Detail",
+                    () => {
+                        if (part.CommentsShown == false)
+                            return null;
+
+                        return shapeHelper.Parts_Comments_Count_Detail(
+                            CommentCount: part.CommentsCount);
+                    }),

```

Comment:
I understand, you want it to be balanced. You can simply change the source code on your machine and push another commit. It will updated this PR automatically.